src/RFC-0313_VNRegistration.md

```diff
@@ -0,0 +1,344 @@
+
+We define a `VNConfirmationPeriod` as is a network-wide constant that specifies the number confirmations (blocks) that a block must have 
+before a validator will recognise it as final. The chosen value for `VNConfirmationPeriod` must be large enough to make reorgs beyond that 
+point practically impossible. Validators simply ignore [base layer] [reorg]s with a depth _less than_ `VNConfirmationPeriod` deep as the 
+data they have extracted is still valid. This means that the _point of finality_ is not always `VNConfirmationPeriod` blocks away from the
+tip and is non-decreasing/monotonic which effectively negates [reorg] "noise" from the chain tip. 
+
+However, this does not address [base layer] latency delays where a single huge block or multi-block reorgs may take seconds to be received 
+and processed by all [base node]s. Moreover, the validators may poll the base layer only every few seconds, further increasing the latency for 
+validators to become aware of the state. This means a single strict validator epoch change-over point will almost always cause liveness failures at 
+and after the epoch transition.
+
+To address this, we define a `VNEpochGracePeriod` where both the previous and current epoch are accepted. This value, in blocks, must allow enough
+time for all validators to become aware of the [base layer] state for the epoch.
+
+To illustrate, consider the following view of a [base layer] chain. We mark 3 views of the chain.
+
+```text
+                                      (a) (b) (c)  (d)              
+                                       |   |   |    |                                    {noisy}
+------ | --x----*------- | --x----*------- | --x----*------- | --x----*------- | --- .... ------> tip
+  |  ϵ10       |   |    ϵ11       |      ϵ12       |       ϵ13               ϵ14
+ V_1          V_2 V_3   V_4      V_5              V_6                      
+Key:
+* - Epoch mid-point
+x - Epoch transition point
+ϵn - Epoch n
+V_n - Validator registrations 
+```
+
+Point (a)
+- the validator node set is incomplete for epoch 12.
+- the active epoch is 11.
+- the validator MUST reject instructions for epoch 12.
+
+Point (b) - the start of epoch 12: 
+- the validator nodeset is final for epoch 12.
+- the active epoch remains at 11.
+- the validator MUST accept instructions from epoch 11 and 12.
+
+Point (c) - the transition point for epoch 12:
+- the active epoch is now 12
+- the validator MUST accept instructions from 11.
+- it is assumed at this point that almost all ($3f + 1$) nodes are now on epoch 12
+
+Point (d) - the epoch mid-point:
+- the active epoch is 12.
+- the validator MUST reject instructions from epoch 11.
+- the validator MUST accept instructions from epoch 12.
+
+#### Validator Node Set Definition
+
+The function $\text{get_vn_set}(\epsilon_n) \rightarrow \vec{S}$ that returns an ordered vector $\vec{S}$ of `VN_Shard_Key`s that are registered for the epoch $\epsilon_n$.
```

Comment:
Not sure if this will show up in the mdbook version, but Github shows an error rendering this:
![image](https://user-images.githubusercontent.com/4200336/201323273-fd33ef7f-5643-402d-8b1f-1099b004f078.png)

Reply:
oh that's annoying, yeah it renders fine in mdbook